**Core -> Full Integration (with SecurityManager)** [Build 271](https://ci.wildfly.org/buildConfiguration/WildFlyCore_PullRequest_WildFlyCoreFullIntegrationLinuxJdk25/574912) outcome was **FAILURE** using a merge of 2cee10661a571a46f19f01daaff43e042cabbe0b
Summary: Tests failed: 1 (1 new), passed: 8159, ignored: 151 Build time: 08:09:29
### Failed tests
```
org.jboss.as.test.integration.hibernate.search.backend.elasticsearch.extension.HibernateSearchElasticsearchGsonDependencyTestCase(basic-integration-default-web): org.jboss.as.test.integration.hibernate.search.backend.elasticsearch.extension.HibernateSearchElasticsearchGsonDependencyTestCase(basic-integration-default-web).: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image docker.elastic.co/elasticsearch/elasticsearch:8.15.4
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:346)
	at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:317)
	at org.jboss.as.test.integration.hibernate.search.backend.elasticsearch.util.ElasticsearchServerSetupObserver.startElasticsearch(ElasticsearchServerSetupObserver.java:65)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)
	at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:331)
	... 44 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:551)
	at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:341)
	at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)
	... 45 more
Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*("message":\s?"started[\s?|"].*|] started
$)'
	at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
	at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:904)
	at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:487)
	... 47 more
------- Stdout: -------
 [33m07:01:06,468 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:37bf has 1 heuristic participant(s)!
 [0m [33m07:01:06,472 WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016037: Could not find new XAResource to use for recovering non-serializable XAResource XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37bf, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37c8, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@311315bf >
 [0m [33m07:01:06,472 WARN  [com.arjuna.ats.jta] (Periodic Recovery) XAResourceRecord restored heuristic instance: XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37bf, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37c8, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@311315bf >
 [0m [33m07:01:06,473 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:37bf restored heuristic participant XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37bf, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37c8, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@311315bf >
 [0m [33m07:01:06,483 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:37f2 has 1 heuristic participant(s)!
 [0m [33m07:01:06,483 WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016037: Could not find new XAResource to use for recovering non-serializable XAResource XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37f2, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37f8, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@1bf99f5b >
 [0m [33m07:01:06,483 WARN  [com.arjuna.ats.jta] (Periodic Recovery) XAResourceRecord restored heuristic instance: XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37f2, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37f8, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@1bf99f5b >
 [0m [33m07:01:06,483 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:37f2 restored heuristic participant XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37f2, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:37f8, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@1bf99f5b >
 [0m [33m07:01:06,487 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:3816 has 1 heuristic participant(s)!
 [0m [33m07:01:06,488 WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016037: Could not find new XAResource to use for recovering non-serializable XAResource XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3816, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:381c, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@326edf76 >
 [0m [33m07:01:06,488 WARN  [com.arjuna.ats.jta] (Periodic Recovery) XAResourceRecord restored heuristic instance: XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3816, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:381c, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@326edf76 >
 [0m [33m07:01:06,488 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:3816 restored heuristic participant XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3816, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:381c, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@326edf76 >
 [0m [33m07:01:06,501 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:381f has 1 heuristic participant(s)!
 [0m [33m07:01:06,501 WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016037: Could not find new XAResource to use for recovering non-serializable XAResource XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:381f, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3825, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@6072e2bb >
 [0m [33m07:01:06,501 WARN  [com.arjuna.ats.jta] (Periodic Recovery) XAResourceRecord restored heuristic instance: XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:381f, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3825, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@6072e2bb >
 [0m [33m07:01:06,501 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:381f restored heuristic participant XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:381f, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3825, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@6072e2bb >
 [0m [33m07:01:06,504 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:3837 has 1 heuristic participant(s)!
 [0m [33m07:01:06,505 WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016037: Could not find new XAResource to use for recovering non-serializable XAResource XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3837, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3840, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@7f9aef4a >
 [0m [33m07:01:06,505 WARN  [com.arjuna.ats.jta] (Periodic Recovery) XAResourceRecord restored heuristic instance: XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3837, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3840, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@7f9aef4a >
 [0m [33m07:01:06,505 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:3837 restored heuristic participant XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3837, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3840, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@7f9aef4a >
 [0m [33m07:01:06,523 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:386a has 1 heuristic participant(s)!
 [0m [33m07:01:06,524 WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016037: Could not find new XAResource to use for recovering non-serializable XAResource XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:386a, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3870, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@28cea4a5 >
 [0m [33m07:01:06,524 WARN  [com.arjuna.ats.jta] (Periodic Recovery) XAResourceRecord restored heuristic instance: XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:386a, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3870, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@28cea4a5 >
 [0m [33m07:01:06,524 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:386a restored heuristic participant XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:386a, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3870, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@28cea4a5 >
 [0m [33m07:01:06,533 WARN  [com.arjuna.ats.arjuna] (Periodic Recovery) Transaction 0:ffffc0a801e4:1cf23ef:6a630a7b:388e has 1 heuristic participant(s)!
 [0m [33m07:01:06,533 WARN  [com.arjuna.ats.jta] (Periodic Recovery) ARJUNA016037: Could not find new XAResource to use for recovering non-serializable XAResource XAResourceRecord < resource:null, txid:< formatId=131077, gtrid_length=29, bqual_length=36, tx_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:388e, node_name=1, branch_uid=0:ffffc0a801e4:1cf23ef:6a630a7b:3894, subordinatenodename=null, eis_name=unknown eis name >, heuristic: TwoPhaseOutcome.HEURISTIC_ROLLBACK com.arjuna.ats.internal.jta.resources.arjunacore.XAResourceRecord@6c9f5ba5 >


```

--- a/core-feature-pack/common/src/main/resources/content/bin/standalone.bat
+++ b/core-feature-pack/common/src/main/resources/content/bin/standalone.bat
@@ -77,9 +77,9 @@ if not %DEBUG_ARG% == "" (
       shift
       set DEBUG_PORT_VAR=%DEBUG_ARG%
    )
-   shift
-   goto READ-ARGS
 )
+shift
+goto READ-ARGS
 
 :READ-VERSION
 set "JAVA_OPTS=-Xmx16m"

--- a/testsuite/scripts/src/test/java/org/wildfly/scripts/test/StandaloneDebugScriptTestCase.java
+++ b/testsuite/scripts/src/test/java/org/wildfly/scripts/test/StandaloneDebugScriptTestCase.java
@@ -1,0 +1,62 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.scripts.test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.junit.Assert;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.wildfly.common.test.ServerHelper;
+
+@RunWith(Parameterized.class)
+public class StandaloneDebugScriptTestCase extends ScriptTestCase {
+    private static final Function<ModelControllerClient, Boolean> STANDALONE_CHECK = ServerHelper::isStandaloneRunning;
+    private static final String DEBUG_LISTEN_PREFIX = "Listening for transport dt_socket at address: ";
+
+    @Parameter(0)
+    public String[] debugArgs;
+
+    @Parameter(1)
+    public String expectedPort;
+
+    public StandaloneDebugScriptTestCase() {
+        super("standalone");
+    }
+
+    @Parameters(name = "{0} -> port {1}")
+    public static Collection<Object[]> data() {
+        return List.of(
+                new Object[] {new String[] {"--debug"}, "8787"},
+                new Object[] {new String[] {"--debug", "9797"}, "9797"},
+                new Object[] {new String[] {"-Dtest=test", "--debug"}, "8787"},
+                new Object[] {new String[] {"-Dtest=test", "--debug", "9797"}, "9797"}
+        );
+    }
+
+    @Override
+    void testScript(final ScriptProcess script) throws InterruptedException, TimeoutException, IOException {
+        final List<String> args = new ArrayList<>(Arrays.asList(ServerHelper.DEFAULT_SERVER_JAVA_OPTS));
+        args.addAll(Arrays.asList(debugArgs));
+        script.start(STANDALONE_CHECK, args.toArray(String[]::new));
+
+        Assert.assertTrue("The process is not running and should be", script.isAlive());
+
+        final var stdout = script.getStdoutAsString();
+        final String expectedMessage = DEBUG_LISTEN_PREFIX + expectedPort;
+        Assert.assertTrue("Expected to find '" + expectedMessage + "' for a server started with " + script.getLastExecutedCmd() + "\nThe server output was:\n" + stdout, stdout.contains(expectedMessage));
+    }
+}
+

